web-client/src/pages/tasks/view.tsx

```diff
@@ -31,18 +30,19 @@ const ShowTaskPage = (props: ShowTaskProps) => {
   const [language, setLanguage] = React.useState("en-US");
 
   async function getCurrentTask() {
-    const jwt = localStorage.getItem(UserConstants.JWT);
-
-    if (!jwt) {
-      return;
-    }
-
     try {
+      let userId = null;
+      const currentUserJson = localStorage.getItem(UserConstants.Current);
+      if (currentUserJson) {
+        const currentUser = JSON.parse(currentUserJson);
+        userId = currentUser.id;
+      }
+
       const response = await axios.get(
         `http://localhost:4000/v1/tasks/view/${props.taskId}`,
         {
           headers: {
-            Authorization: jwt,
+            "User-ID": userId,
```

Comment:
I think the `checkJwt` function does some smart stuff and you can just access `req.jwtPayload`. Oh but that one needs you logged in.

So you should copy that one and make a `maybeCheckJwt` I guess that doesn't return a 401 if there's no token and also doesn't set `req.jwtPayload`

Reply:
This technique isn't secure because some dude can just figure out Kevin's user id and start looking at secret problems. :P